fiftyone/core/dataset.py

```diff
@@ -966,6 +967,31 @@ def _add_samples_batch(self, samples, expand_schema):
 
         return [str(d["_id"]) for d in dicts]
 
+    def _bulk_update(self, sample_ids, updates, ordered=False):
```

Comment:
@benjaminpkane not sure if you have any reservations about this, or better ideas for how to set a field on each sample of a dataset in cases where the value can't be computed purely in an aggregation pipeline (in which case `set_field()` can be used for this).

I found `bulk_write()` to be significantly faster than the loop I commented out here.

Reply:
No issue here. At some point it would be nice to ensure that only generators are used for `bulk_write` and `insert_many`, but that is a scale issue.